### PR TITLE
bugfix: fix when TC returns RollbackFailed  FailureHandler not executed

### PR DESCRIPTION
**Pull request by ZhangShiYeChina**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
fixes :https://github.com/seata/seata/issues/5231

### CODE REVIEW

File: tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java [hunk 1]
```diff
@@ -239,9 +239,29 @@ private void rollbackTransaction(GlobalTransaction tx, Throwable originalExcepti
...
+                code = TransactionalExecutor.Code.RollbackFailure;
+                break;
+            case Rollbacking:
+            case RollbackRetrying:
+                code = TransactionalExecutor.Code.RollbackRetrying;
+                break;
+            case TimeoutRollbacking:
+            case TimeoutRollbackRetrying:
+            case TimeoutRollbacked:
+            case RollbackRetryTimeout:
```
Comment (a364176773) [inline tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java:256]: RollbackRetryTimeout 应该放在RollbackFailure那块，因为他是一个失败状态，且不会进行重试了

Comment (ZhangShiYeChina) [reply tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java:256]: 好的，我修改一下

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**